src/ray/core_worker/task_manager.cc

```diff
@@ -456,22 +462,35 @@ bool TaskManager::RetryTaskIfPossible(const TaskID &task_id) {
         << "Tried to retry task that was not pending " << task_id;
     spec = it->second.spec;
     num_retries_left = it->second.num_retries_left;
-    if (num_retries_left > 0) {
-      it->second.num_retries_left--;
+    num_oom_retries_left = it->second.num_oom_retries_left;
+    if (task_failed_due_to_oom) {
+      if (num_oom_retries_left > 0) {
+        will_retry = true;
+        it->second.num_oom_retries_left--;
+      }
```

Comment:
so here if task failed due to OOM and num oom retry exhausted, we don't retry even if normal task retry is non zero. Is this intentional?

Reply:
Yea that's what I implemented in the original PR. But seems we decided to make this simple per Stephanie's comment here (and got a thumbs up) 

https://github.com/clarng/ray/pull/1#discussion_r978759893